packages/typespec-ts/src/modular/buildOperations.ts

```diff
@@ -53,33 +56,33 @@ export function buildOperationFiles(
         moduleSpecifier: "@azure-rest/core-client",
         namedImports: [
           "StreamableMethod",
-          "operationOptionsToRequestParameters",
-          "OperationOptions"
+          "operationOptionsToRequestParameters"
         ]
       }
     ]);
+    modelOptionsFile.addImportDeclarations([
+      {
+        moduleSpecifier: "@azure-rest/core-client",
+        namedImports: ["OperationOptions"]
+      }
+    ]);
 
-    // Import models used from ./models.ts
-    importModels(operationGroupFile, project);
-    operationGroupFile.fixMissingImports();
-  }
-}
-
-function importModels(sourceFile: SourceFile, project: Project) {
-  const modelsFile = project.getSourceFile("models.ts");
-  const models: string[] = [];
+    modelOptionsFile.fixMissingImports();
+    modelOptionsFile
+      .getImportDeclarations()
+      .filter((id) => {
+        return (
+          id.isModuleSpecifierRelative() &&
+          !id.getModuleSpecifierValue().endsWith(".js")
+        );
+      })
+      .map((id) => {
+        id.setModuleSpecifier(id.getModuleSpecifierValue() + ".js");
+        return id;
+      });
 
-  for (const entry of modelsFile?.getExportedDeclarations().entries() ?? []) {
-    models.push(entry[0]);
+    operationGroupFile.fixMissingImports();
   }
-
-  sourceFile.addImportDeclaration({
-    moduleSpecifier: "./models.js",
-    namedImports: models
-  });
-
-  // Import all models and then let ts-morph clean up the unused ones
-  sourceFile.fixUnusedIdentifiers();
```

Comment:
Remove this because this is actually not working, because by the time we generate operations.ts file, models folder has not been ready yet. and we use ts-morph `operationGroupFile.fixMissingImports();` which is smart enough to import those dependencies.